ako-gateway-api/k8s/validator.go

```diff
@@ -376,6 +382,12 @@ func validateParentReference(key string, httpRoute *gatewayv1.HTTPRoute, httpRou
 		defaultCondition.
 			Message(err.Error()).
 			SetIn(&httpRouteStatus.Parents[index].Conditions)
+		found, hosts := akogatewayapiobjects.GatewayApiLister().GetGatewayRouteToHostname(namespace, name)
```

Comment:
@akshayhavile This is for the scenario where we had hosts in httproute that matched the gateway listeners but later the httproute was updated with hosts that do not match the gateway listener.  **found** will be true in this case since we had valid matching hosts previously. Our goal is to delete the child vs and associated pg/pool if the above scenario happens. So, if we do not return nil from here, then the httproute will be marked invalid and the existing child vs etc associated with the httproute will not be deleted.